migration/process-migration/src/test/java/io/camunda/migration/process/it/es/ElasticsearchMigrationRunnerIT.java

```diff
@@ -0,0 +1,510 @@
+    assertThat(records.size()).isEqualTo(9);
+    assertThat(records.stream().filter(r -> r.getKey() <= 5).allMatch(r -> r.getIsPublic() == null))
+        .isTrue();
+    assertThat(records.stream().filter(r -> r.getKey() <= 5).allMatch(r -> r.getFormId() == null))
+        .isTrue();
+    assertThat(records.stream().filter(r -> r.getKey() <= 5).allMatch(r -> r.getFormKey() == null))
+        .isTrue();
+    assertThat(
+            records.stream()
+                .filter(r -> r.getKey() <= 5)
+                .allMatch(r -> r.getIsFormEmbedded() == null))
+        .isTrue();
+
+    assertThat(
+            records.stream()
+                .filter(r -> r.getKey() > 5)
+                .allMatch(r -> r.getIsPublic().equals(Boolean.TRUE)))
+        .isTrue();
+    assertThat(
+            records.stream()
+                .filter(r -> r.getKey() > 5)
+                .allMatch(r -> r.getFormId().equals("testForm")))
+        .isTrue();
+    assertThat(records.stream().filter(r -> r.getKey() > 5).allMatch(r -> r.getFormKey() == null))
+        .isTrue();
+    assertThat(
+            records.stream()
+                .filter(r -> r.getKey() > 5)
+                .allMatch(r -> r.getIsFormEmbedded().equals(Boolean.FALSE)))
+        .isTrue();
+  }
+
+  @Test
+  public void shouldNotMigrateWhenFinalStepIsPresent() throws IOException {
+    // given
+    properties.setBatchSize(1);
+    final ProcessEntity entityToBeMigrated = TestData.processEntityWithPublicFormId(1L);
+    final ProcessEntity entityNotToBeMigrated = TestData.processEntityWithPublicFormId(2L);
+    writeImportPositionToIndex(TestData.completedImportPosition(1));
+    writeProcessorStepToIndex("2");
+    writeProcessToIndex(entityToBeMigrated);
+    writeProcessToIndex(entityNotToBeMigrated);
+    awaitRecordsArePresent(ProcessEntity.class, processIndex.getFullQualifiedName());
+
+    // when
+    migrator.run();
+    esClient.indices().refresh();
+
+    // then
+    final var records = readRecords(ProcessEntity.class, processIndex.getFullQualifiedName());
+    final var stepRecords =
+        readRecords(ProcessorStep.class, migrationRepositoryIndex.getFullQualifiedName());
+    assertThat(records.size()).isEqualTo(2);
+    assertThat(records.stream().allMatch(r -> r.getIsPublic() == null)).isTrue();
+    assertThat(records.stream().allMatch(r -> r.getFormId() == null)).isTrue();
+    assertThat(records.stream().allMatch(r -> r.getFormKey() == null)).isTrue();
+    assertThat(records.stream().allMatch(r -> r.getIsFormEmbedded() == null)).isTrue();
+    assertThat(stepRecords.size()).isEqualTo(1);
+    assertThat(stepRecords.getFirst().getContent()).isEqualTo("2");
+  }
+
+  @Test
+  public void shouldMigrateDuringCountdown() throws IOException {
+    // given
+    properties.setImporterFinishedTimeout(Duration.ofSeconds(2));
+    properties.setMinRetryDelay(Duration.ofSeconds(3));
+    properties.setMaxRetryDelay(Duration.ofSeconds(3));
+    properties.setBatchSize(4);
```

Comment:
🔧 Lets try to find smaller numbers for this or maybe a different way to control the behavior, like controllable clock or something 🤔